### PR TITLE
PXS-587 Implement Hilt & Mock Test Firestore

### PR DESCRIPTION
**Pull request by qtipee**
- Implement Hilt in order to do dependency injection
- Add the MockFirebaseExperienceRepository

### CODE REVIEW

**Comment by xavierpantet** [comment]
@timotheeduran please review this as well!

You should probably have a look [here](https://pindex.atlassian.net/wiki/spaces/PI/pages/711327745/Dependency+Injection).

File: app/src/main/java/com/pindex/main/home/ExperienceActivity.kt [hunk 1]
```diff
@@ -11,6 +11,7 @@ import com.pindex.main.models.ExperienceDto
 import com.pindex.main.ui.fragments.blocks.*
 import com.pindex.main.utils.Constants
 
+//@AndroidEntryPoint
```
Comment (xavierpantet) [inline app/src/main/java/com/pindex/main/home/ExperienceActivity.kt:14]: Remove comment

File: app/src/main/java/com/pindex/main/home/HomeActivity.kt [hunk 2]
```diff
@@ -15,4 +17,4 @@ class HomeActivity : AppCompatActivity() {
         PindexFirebase.signInAnonymously(this)
```
Comment (xavierpantet) [inline app/src/main/java/com/pindex/main/home/HomeActivity.kt:17]: Is it possible to sign in to Firebase only when we instantiate the repository?

Maybe in `ExperienceRepositoryDI`?

File: app/src/main/java/com/pindex/main/repositories/FirebaseMockExperienceRepository.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Data Faker
+     */
+    private val faker = Faker()
+
+    override suspend fun getPage(limit: Long): List<ExperienceDto> {
+        val experiencesList = ArrayList<ExperienceDto>()
+
+        // Create [limit] experiences and add them to the list
+        for (i in 1..limit) {
```
Comment (xavierpantet) [inline app/src/main/java/com/pindex/main/repositories/FirebaseMockExperienceRepository.kt:21]: I think we can be more concise:

```kotlin
return Array(limit, { _ -> createRandomExperience() }).toCollection(ArrayList())
```

File: app/src/main/java/com/pindex/main/repositories/FirebaseMockExperienceRepository.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        return experiencesList
+    }
+
+    /**
+     * Create and return an ExperienceDto, whose short description
+     * and title are random strings.
+     */
+    private fun createRandomExperience(): ExperienceDto {
+        val contentList = ArrayList<ContentDto>()
+        val content = ContentDto(
```
Comment (xavierpantet) [inline app/src/main/java/com/pindex/main/repositories/FirebaseMockExperienceRepository.kt:35]: Same comment here on conciseness

File: app/src/main/java/com/pindex/main/services/ExperienceService.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import com.pindex.main.models.ExperienceDto
+import com.pindex.main.repositories.ExperienceRepository
+
+/**
+ * Experience Service for the Pindex experiences. This class needs
+ * a repository in order to get the paginated data.
+ */
+class ExperienceService (private val repository: ExperienceRepository) {
+    /*
+    class ExperienceService @Inject constructor (
```
Comment (xavierpantet) [inline app/src/main/java/com/pindex/main/services/ExperienceService.kt:12]: Remove comment

**Review by xavierpantet — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pindex/main/home/HomeActivity.kt (hunk 2 above)
Comment (qtipee) [reply app/src/main/java/com/pindex/main/home/HomeActivity.kt:17]: I moved the code that logs the user in inside the `FirebaseExperienceRepository` and it seems to work!

**Comment by xavierpantet** [comment]
Looks good to me! GG @qtipee!

**Review by timotheeduran — APPROVED** [review]

**Review by xavierpantet — APPROVED** [review]

**Pull request merged**